hcode-hw062823a.opmst failed to compile with Fedora 39 ppc64le

## Info

I tried to build a RaptorCS blackbird firmware https://github.com/open-power/op-build/pull/5256 on Fedora 39 ppc64le but got error with hcode-hw062823a.opmst (see **Error Output**)

I have tried GCC 8.4.0, 9.4, 10.4.0 and 12.3.0 but still produce the same error.

# System Details Report
---

## Report details
- **Date generated:**                              2023-08-20 06:58:21

## Hardware Information:
- **Hardware Model:**                        Raptor Computing Blackbird C1P9S01 REV 1.01
- **Memory:**                                      61.4 GiB
- **Processor:**                                   POWER9, altivec supported × 32
- **Graphics:**                                    AMD Radeon™ RX 6600 XT
- **Disk Capacity:**                               8.0 TB

## Software Information:
- **OS Name:**                                     Fedora Linux 39 (Workstation Edition Prerelease)
- **OS Arch:**                                    PowerPC 64 LE
- **OS Type:**                                     64-bit
- **Kernel Version:**                              Linux 6.5.0-0.rc6.20230816git4853c74bd7ab.45.fc39.ppc64le

## Error Output

``` 
    p9_sgpe_img_edit.o
    CC         p9_dd_container.o
    AR         libcen_ringId.a
    C++        p9_scan_compression.o
    AR         libstop_gpeImgEdit.a
    C++        p9_tor.o
    AR         libp9_ring_identification.a
    AR         libp9_xip_image.a
    LD         pstate_gpeImgEdit.exe
    AR         libp9_dd_container.a
    LD         cmeImgEdit.exe
    LDSO       libp9_ringId.so
    AR         libp9_tor.a
    LDSO       libcen_ringId.so
    LD         stop_gpeImgEdit.exe
    C++        common_ringId.o
    LDSO       libp9_ring_identification.so
    LDSO       libp9_dd_container.so
    LDSO       libp9_xip_image.so
    AR         libcommon_ringId.ahcode-hw062823a
    LDSO       libp9_tor.so
    LDSO       libcommon_ringId.so
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C: In function ‘int rs4_embed_cmsk(CompressedScanData**, CompressedScanData*)’:
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C:935:26: error: pointer ‘embeddedAddr’ may be used after ‘void* realloc(void*, size_t)’ [-Werror=use-after-free]
  935 |     memmove(embeddedAddr + embeddedSize,
      |             ~~~~~~~~~~~~~^~~~~~~~~~~~~~
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C:927:43: note: call to ‘void* realloc(void*, size_t)’ here
  927 |     *io_rs4 = (CompressedScanData*)realloc(*io_rs4, totalSize);
      |                                    ~~~~~~~^~~~~~~~~~~~~~~~~~~~
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C:940:11: error: pointer ‘embeddedAddr’ may be used after ‘void* realloc(void*, size_t)’ [-Werror=use-after-free]
  940 |     memcpy(embeddedAddr,
      |     ~~~~~~^~~~~~~~~~~~~~
  941 |            i_rs4Cmsk,
      |            ~~~~~~~~~~
  942 |            embeddedSize);
      |            ~~~~~~~~~~~~~
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C:927:43: note: call to ‘void* realloc(void*, size_t)’ here
  927 |     *io_rs4 = (CompressedScanData*)realloc(*io_rs4, totalSize);
      |                                    ~~~~~~~^~~~~~~~~~~~~~~~~~~~
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C:935:12: error: pointer ‘embeddedAddr’ may be used after ‘void* realloc(void*, size_t)’ [-Werror=use-after-free]
  935 |     memmove(embeddedAddr + embeddedSize,
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  936 |             embeddedAddr,
      |             ~~~~~~~~~~~~~
  937 |             be16toh((*io_rs4)->iv_size) - sizeof(CompressedScanData));
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../import/chips/p9/utils/imageProcs//p9_scan_compression.C:927:43: note: call to ‘void* realloc(void*, size_t)’ here
  927 |     *io_rs4 = (CompressedScanData*)realloc(*io_rs4, totalSize);
      |                                    ~~~~~~~^~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[3]: *** [../../import/chips/p9/utils/imageProcs/p9_scan_compression.mk:32: ../../import/../output/obj/p9_scan_compression/p9_scan_compression.o] Error 1
make[2]: *** [common.dir/order.final.mk:41: _BUILD/MODULE_TARGETS] Error 2
make[1]: *** [Makefile:37: all] Error 2
make[1]: Leaving directory '/home/tle/Work/op-build/output/build/hcode-hw062823a.opmst'
make: *** [package/pkg-gene
```

--- a/import/chips/p9/utils/imageProcs/p9_scan_compression.C
+++ b/import/chips/p9/utils/imageProcs/p9_scan_compression.C
@@ -919,7 +919,6 @@ int
 rs4_embed_cmsk( CompressedScanData** io_rs4,
                 CompressedScanData*  i_rs4Cmsk )
 {
-    char* embeddedAddr = (char*)(*io_rs4 + 1);
     size_t embeddedSize = be16toh(i_rs4Cmsk->iv_size);
     size_t totalSize   = be16toh((*io_rs4)->iv_size) + embeddedSize;
 
@@ -930,6 +929,10 @@ rs4_embed_cmsk( CompressedScanData** io_rs4,
     {
         return BUG(SCAN_COMPRESSION_NO_MEMORY);
     }
+
+    // realloc() above may move the buffer, so derive the embedded address from
+    // the (possibly new) *io_rs4 pointer rather than from the pre-realloc one.
+    char* embeddedAddr = (char*)(*io_rs4 + 1);
 
     // Make space for cmsk ring
     memmove(embeddedAddr + embeddedSize,

